### PR TITLE
Integrate upgrade tests (golang+cucushift)

### PR DESCRIPTION
**Pull request by liangxia**
Running upgrade tests ( golang + cucushift ) in a unified place.

### CODE REVIEW

File: ci-operator/step-registry/openshift/upgrade/test/openshift-upgrade-test-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+chain:
+  as: openshift-upgrade-test
+  steps:
+  - ref: cucushift-pre
+  - ref: openshift-extended-upgrade-pre
+  - ref: cucushift-upgrade-prepare
+  - ref: cucushift-upgrade-prehealthcheck
+  - ref: cucushift-upgrade-toimage
+  - ref: cucushift-upgrade-healthcheck
```
Comment (jiajliu) [inline ci-operator/step-registry/openshift/upgrade/test/openshift-upgrade-test-chain.yaml:9]: `healthcheck` was already integrated into `cucushift-upgrade-toimage`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml [hunk 2]
```diff
@@ -63,20 +63,9 @@ tests:
...
-    - ref: openshift-extended-upgrade-pre
-    - ref: cucushift-pre
-    - ref: cucushift-upgrade-prepare
-    - ref: cucushift-upgrade-prehealthcheck
-    - ref: cucushift-upgrade-toimage
-    - ref: cucushift-upgrade-healthcheck
-    - ref: openshift-extended-upgrade-post
-    - ref: cucushift-upgrade-check
-    - ref: cucushift-upgrade-report
+    - chain: openshift-upgrade-test
```
Comment (jiajliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:67]: This chain `openshift-upgrade-test` seems for golang+cucushift cases. And there is another upgrade chain `https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml` for cucushift only. I wonder if we need both of them? if not, maybe we need integrate them into one. if yes, we hold them into two dirs by purpose?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:67]: Once we finished migrating existing upgrade jobs, we can remove cucushift-upgrade-chain

Comment (jiajliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:67]: ack

File: ci-operator/step-registry/openshift/upgrade/test/OWNERS [hunk 3]
```diff
@@ -0,0 +1,7 @@
+approvers:
```
[the remark this answers is not in the document]
Comment (liangxia) [reply ci-operator/step-registry/openshift/upgrade/test/OWNERS:1]: It's a generic workflow to run upgrade test on all supported platforms.

[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/step-registry/openshift/upgrade/test/OWNERS:1]: Hi Liang,
Seems like it's a QE-specific workflow. So would it be better to change it to something like qe-test to distinguish the existing dev-specific workflows?

[the remark this answers is not in the document]
Comment (liangxia) [reply ci-operator/step-registry/openshift/upgrade/test/OWNERS]: Moving those content into a sub-folder qe.
Does it make sense ?

**Review by wking — APPROVED** [review]
/lgtm

**Pull request merged**